cuML/src/dbscan/labelling/naive.h

```diff
@@ -55,45 +57,41 @@ void bfs(int id, Type *host_adj_graph,
 }
 
 template <typename Type>
-void launcher(Pack<Type> data, int startVertexId, int batchSize, cudaStream_t stream) {
+void launcher(const ML::cumlHandle_impl& handle, Pack<Type> data, int startVertexId, int batchSize, cudaStream_t stream) {
     size_t N = (size_t)data.N;
-    Type *host_vd = new Type[N+1];
-    bool *host_core_pts = new bool[N];
-    bool *host_visited = new bool[N];
-    Type *host_ex_scan = new Type[N];
-    Type *host_db_cluster = new Type[N];
-    bool *host_xa = new bool[N]();
+    MLCommon::host_buffer<Type> host_vd(handle.getHostAllocator(), stream, sizeof(Type)*(N+1));
```

Comment:
Size of `host_buffer` is in terms of number of elements. 6 more instances below.